src/object.c

```diff
@@ -958,6 +961,11 @@ int compareStringObjectsWithFlags(const robj *a, const robj *b, int flags) {
     }
     if (flags & REDIS_COMPARE_COLL) {
         return strcoll(astr,bstr);
+    } else if (flags & REDIS_COMPARE_EQUAL) {
+        if (alen != blen) {
+            return 0;
```

Comment:
now, if a == b returns 1, unequal returns 0, but in the original code it should be:
<0 	 	a < b
==0 	a == b
>0 		a > b

Please confirm that the test correctly covers it.

Reply:
What I mean is that the intention of the return value has been changed. Originally, there were 3 types of return values, but now there are only 0 and 1.

Reply:
Well, maybe we should just handle this case inside equalStringObjects and keep compareStringObjectsWithFlags unchanged. I’ll make the changes later. Marking the conversation as resolved. Thanks

Reply:
@vitahlin sorry, what I said yesterday about adding tests refers to why the meaning of the return value changes but runtest doesn't fail. I think it's not necessary to add so many unittests for it.

Reply:
Revert the commit of unit test